test/diamond/DiamondERC721.test.ts

```diff
@@ -0,0 +1,338 @@
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
+import { expect } from "chai";
+import { Reverter } from "@/test/helpers/reverter";
+import { getSelectors, FacetAction } from "@/test/helpers/diamond-helper";
+import { ZERO_ADDR } from "@/scripts/utils/constants";
+
+import { OwnableDiamondMock, DiamondERC721Mock, Diamond } from "@ethers-v6";
+
+describe("DiamondERC721 and InitializableStorage", () => {
+  const reverter = new Reverter();
+
+  let OWNER: SignerWithAddress;
+  let SECOND: SignerWithAddress;
+  let THIRD: SignerWithAddress;
+
+  let erc721: DiamondERC721Mock;
+  let diamond: OwnableDiamondMock;
+
+  before("setup", async () => {
+    [OWNER, SECOND, THIRD] = await ethers.getSigners();
+
+    const OwnableDiamond = await ethers.getContractFactory("OwnableDiamondMock");
+    const DiamondERC721Mock = await ethers.getContractFactory("DiamondERC721Mock");
+
+    diamond = await OwnableDiamond.deploy();
+    erc721 = await DiamondERC721Mock.deploy();
+
+    const facets: Diamond.FacetStruct[] = [
+      {
+        facetAddress: await erc721.getAddress(),
+        action: FacetAction.Add,
+        functionSelectors: getSelectors(erc721.interface),
+      },
+    ];
+
+    await diamond.diamondCutShort(facets);
+
+    erc721 = <DiamondERC721Mock>DiamondERC721Mock.attach(await diamond.getAddress());
+
+    await erc721.__DiamondERC721Mock_init("Mock Token", "MT");
+
+    await reverter.snapshot();
+  });
+
+  afterEach(reverter.revert);
+
+  describe("access", () => {
+    it("should initialize only once", async () => {
+      await expect(erc721.__DiamondERC721Mock_init("Mock Token", "MT")).to.be.revertedWith(
+        "Initializable: contract is already initialized"
+      );
+    });
+
+    it("should initialize only by top level contract", async () => {
+      await expect(erc721.__DiamondERC721Direct_init("Mock Token", "MT")).to.be.revertedWith(
+        "Initializable: contract is not initializing"
+      );
+    });
+
+    it("should disable implementation initialization", async () => {
```

Comment:
Does this test actually test anything? It seems like it tests the mock contract...